### PR TITLE
Implement useReducer for textAreaValues

### PR DESCRIPTION
**Pull request by noi5e**
Hi again everyone! 👋🏽 I'd like to spend a couple of weeks finishing up a solid React rewrite of the commenting system! Toward the end, I'm going to write a bunch of planning issues so that other contributors can continue the work as needed.

Picking up from where I left off, I'm trying to simplify the app's `state` by using React's `useReducer` feature.

---
(This issue is part of the larger Comment Editor Overhaul Project with Outreachy. Refer to Planning Issues #9069 and #9365 for more context)

### CODE REVIEW

File: app/javascript/components/CommentsContainer.js [hunk 1]
```diff
@@ -55,17 +61,28 @@ const CommentsContainer = ({
...
+          commentFormId: formId,
+          newValue: ""
+        });
+        dispatch({
+          type: "CREATE NEW TEXTAREA VALUE",
+          commentFormId: "edit-" + newCommentId,
+          newValue: newCommentRawText
+        });
         // the new comment form comes with an edit form, its toggle state needs to be created as well
+        // TODO: create multicase for "CREATE NEW COMMENT FORM" in reducers.js
```
[the remark this answers is not in the document]
Comment (noi5e) [reply app/javascript/components/CommentsContainer.js:76]: Seems inapplicable because this is within a comment! Possible to skip?

**Comment by jywarren** [comment]
Hi @noi5e its great to hear from you! 🎉 This looks wonderful and many thanks.

**Comment by jywarren** [comment]
🎉 🎉 🎉

**Comment by noi5e** [comment]
Thanks @jywarren!

**Pull request merged**